src/lib.rs

```diff
@@ -315,9 +315,6 @@ impl Litep2p {
 
             for address in transport_listen_addresses {
                 transport_manager.register_listen_address(address.clone());
-                listen_addresses.push(address.with(Protocol::P2p(
```

Comment:
Here and below we should keep old listen addresses separate from public addresses.